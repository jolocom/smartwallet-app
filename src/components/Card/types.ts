import { StyleProp, ViewStyle } from 'react-native'
import { ClaimEntry } from 'jolocom-lib/js/credentials/credential/types'

export interface IWithCustomStyle<T = ViewStyle> {
  customStyles?: StyleProp<T>
}

export interface IField {
  label: string
  value: ClaimEntry
}

export interface ICardProps {
  id: string
  type: string
  optionalFields: IField[]
  mandatoryFields: Array<IField | null>
  photo?: string | undefined
  highlight?: string | undefined
}

interface IOptionalFields extends IWithCustomStyle {
  lastFieldPadding?: string
}

export interface ICardComposition {
  OptionalFields: React.FC<IOptionalFields>
  DocumentHeader: React.FC
  OtherHeader: React.FC
  Highlight: React.FC
  Photo: React.FC
  Dots: React.FC<IWithCustomStyle>
}
