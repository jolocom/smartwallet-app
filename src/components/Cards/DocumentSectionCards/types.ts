import { DisplayVal } from '@jolocom/sdk/js/credentials'

interface CommonDocumentsProps {
  credentialName: string
  fields: Array<Required<DisplayVal>>
  onHandleMore: () => void
}

export interface OtherCardProps extends CommonDocumentsProps {
  credentialType: string
  logo?: string
}

export interface DocumentCardProps extends CommonDocumentsProps {
  holderName: string
  highlight?: string
  photo?: string
}
