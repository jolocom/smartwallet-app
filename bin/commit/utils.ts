import childProcess from 'child_process'

export const promisify =
  <T>(fn: (args: T) => void) =>
  (args: T): Promise<void> =>
    new Promise((res, rej) => {
      try {
        fn(args)
        res()
      } catch (e) {
        rej(e)
      }
    })

export const listStagedFiles = () =>
  childProcess.execSync(`git diff --cached --name-only --diff-filter=ACMR`)

export const stageModifiedFiles = (stagedFiles: string) =>
  childProcess.execSync(`echo "${stagedFiles}" | xargs git add`)

export const abortScript = (msg: string) => {
  console.log('')
  console.log('\x1b[0;33m%s\x1b[0m', msg)
  console.log('\x1b[0;31m%s\x1b[0m', 'Aborting')
  process.exit(1)
}

export const logStep = (msg: string) => {
  console.log('\x1b[0;34m%s\x1b[0m', `${msg}`)
}

export const spawnProcess = (
  onClose: (
    code: number | null,
    dataOutput: string,
    errorOutput: string,
  ) => void,
  cmd: string,
  args?: readonly string[],
  options?: childProcess.SpawnOptionsWithoutStdio,
) => {
  const spawnedProcess = childProcess.spawn(cmd, args, options)

  let dataOutput = ''
  let errorOutput = ''

  spawnedProcess.stdout.on('data', (data) => {
    dataOutput += data.toString() + '\n'
  })
  spawnedProcess.stderr.on('data', (error: Buffer) => {
    errorOutput += error.toString() + '\n'
  })

  spawnedProcess.stderr.pipe(process.stderr)
  spawnedProcess.stdout.pipe(process.stdout)

  spawnedProcess.on('close', (code) => onClose(code, dataOutput, errorOutput))

  return spawnedProcess
}
