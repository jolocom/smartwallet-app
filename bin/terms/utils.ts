import fetch from 'node-fetch';
import fs from 'fs';
import shell from 'shelljs'

const CURRENT_PATH = process.cwd();
const SECRETS_REPO = 'dev@hetz1.jolocom.io:poeditor-key'
const BASE_URL = 'https://api.poeditor.com/v2';
const API_KEY_LOCATION = `${CURRENT_PATH}/bin/poeditor-key/POEDITOR_TOKEN.txt`
const PROJECT_ID_LOCATION = `${CURRENT_PATH}/bin/poeditor-key/POEDITOR_SW2_PROJECT_ID.txt`

export enum Languages {
  en = 'en',
  de = 'de',
}

interface IResult  {
  [key: string]: {
    [key: string]: number
  }
}

export interface IResponse<T = IResult> {
  response: {
    status: string,
    code: string,
    message: string,
  },
  result: T
}

export const cloneSecrets = () => {
  return new Promise((res, rej) => {
    const keyExists = fs.existsSync(API_KEY_LOCATION);
    const idExists = fs.existsSync(PROJECT_ID_LOCATION);
    if(!keyExists || !idExists) {
      shell.cd(`${CURRENT_PATH}/bin`)
      shell.rm('-rf', `poeditor-key`)
      shell.exec(
        `git clone ${SECRETS_REPO} -b master`,
        {},
        (code, stdout, stderr) => {
          if (code != 0) return rej(new Error(stderr))
          return res(stdout)
        },
      )
    } else {
      res('Secrets are already present');
    }
  })
}

const readTextFile = (location: string) => {
  return new Promise<string>((res, rej) => {
    fs.readFile(location, 'utf8', (err, data) => {
      if (err) rej(err)
      else res(data.toString().trim())
    })
  })
}

const getPoeditorSecrets = async () => {
    const apiKey = await readTextFile(API_KEY_LOCATION)
    const projectId = await readTextFile(PROJECT_ID_LOCATION)
    return { apiKey, projectId }
}


export const sendPostRequest = async <T = IResponse, >(path: string, params: Record<string, string>) => {
  try {
    const { apiKey, projectId } = await getPoeditorSecrets();
    const requestParams = new URLSearchParams({
      api_token: apiKey,
      id: projectId,
      ...params
    })
    const res = await fetch(`${BASE_URL}${path}`, {
      method: 'POST',
      body: requestParams.toString(),
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded'
      }
    })
    return await res.json() as T;
  } catch(err) {
    console.error(err);
  }
}