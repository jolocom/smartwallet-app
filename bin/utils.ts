import fs from 'fs'
import { stringify as stringifyEnv, parse as parseEnv, Data } from 'envfile'
import shell from 'shelljs'

export const cloneRepo = async (remote: string, repoName: string) => {
  return new Promise<string>((res, rej) => {
    const dir = shell.tempdir()
    console.log(`${dir}/${repoName}`)
    shell.rm('-rf', `${dir}/${repoName}`)
    shell.exec(
      `git clone ${remote}:${repoName} -b master ${dir}/${repoName}`,
      {},
      (code, stdout, stderr) => {
        if (code != 0) return rej(new Error(stderr))
        return res(`${dir}/${repoName}`)
      },
    )
  })
}

export const catFile = async (path: string) => {
  return new Promise((res, rej) => {
    const dsn = shell.cat(path)
    if (dsn.stderr) rej(dsn.stderr)
    res(dsn.stdout)
  })
}

export const writeEnv = async (path: string, data: object) => {
  const env = stringifyEnv(data)

  return new Promise((res, rej) => {
    fs.writeFile(path, env, (err) => {
      if (err) rej(err)
      else res(undefined)
    })
  })
}

export const readEnv = async (path: string) => {
  return new Promise<Data>((res, rej) => {
    fs.readFile(path, 'utf8', (err, data) => {
      if (err) {
        console.log(
          '\x1b[4;31m%s\x1b[0m',
          'Error! Could not find the environmental config (.env).',
        )
      } else res(parseEnv(data))
    })
  })
}
