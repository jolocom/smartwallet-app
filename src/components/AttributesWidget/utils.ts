import { AttrsState } from '~/modules/attributes/types'
import { SelectableAttrI } from './index'

export const getMappedAttrubited = (attrs: AttrsState<string>) =>
  Object.keys(attrs).reduce((acc, val) => {
    acc[val] = attrs[val].map((attr) => ({
      val: attr,
      isSelected: false,
    }))
    return acc
  }, {} as AttrsState<SelectableAttrI>)
