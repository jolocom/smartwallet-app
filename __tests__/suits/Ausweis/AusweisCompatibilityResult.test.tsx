import { fireEvent } from '@testing-library/react-native'
import React from 'react'
import { useRoute } from '@react-navigation/native'
import { useGoBack } from '~/hooks/navigation'
import { AusweisCompatibilityResult } from '~/screens/Modals/Interaction/eID/components'
import { eIDScreens } from '~/screens/Modals/Interaction/eID/types'
import { renderWithSafeArea } from '../../utils/renderWithSafeArea'

jest.mock('@react-navigation/native')
jest.mock('../../../src/hooks/navigation')

describe('Ausweis compatibility result screen', () => {
  const mockGoBack = jest.fn()
  describe('is according to the designs when:', () => {
    beforeAll(() => {
      ;(useGoBack as jest.Mock).mockReturnValue(mockGoBack)
    })
    afterEach(() => {
      ;(useRoute as jest.Mock).mockClear()
    })
    afterAll(() => {
      mockGoBack.mockClear()
    })

    test('scanned card is fully supported', () => {
      ;(useRoute as jest.Mock).mockReturnValue({
        params: {
          inoperative: false,
          deactivated: false,
        },
      })
      const { toJSON } = renderWithSafeArea(
        <AusweisCompatibilityResult
          key="123"
          name={eIDScreens.CompatibilityResult}
          params={{
            inoperative: false,
            deactivated: false,
          }}
        />,
      )
      expect(toJSON()).toMatchSnapshot()
    })

    test('scanned card is inopeartive', () => {
      ;(useRoute as jest.Mock).mockReturnValue({
        params: {
          inoperative: true,
          deactivated: false,
        },
      })
      const { toJSON } = renderWithSafeArea(<AusweisCompatibilityResult />)
      expect(toJSON()).toMatchSnapshot()
    })

    test('scanned card is deactivated', () => {
      ;(useRoute as jest.Mock).mockReturnValue({
        params: {
          inoperative: false,
          deactivated: true,
        },
      })
      const { toJSON } = renderWithSafeArea(<AusweisCompatibilityResult />)
      expect(toJSON()).toMatchSnapshot()
    })
  })

  test('can be dismissed', () => {
    ;(useRoute as jest.Mock).mockReturnValue({
      params: {
        inoperative: false,
        deactivated: false,
      },
    })
    const { getByTestId } = renderWithSafeArea(<AusweisCompatibilityResult />)
    const dismissableArea = getByTestId('dismissable-background')
    fireEvent.press(dismissableArea)
    expect(mockGoBack).toBeCalledTimes(1)
  })
})
