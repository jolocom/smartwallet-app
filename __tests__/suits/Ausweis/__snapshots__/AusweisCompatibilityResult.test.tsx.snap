// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ausweis compatibility result screen is according to the designs when: scanned card is deactivated 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
        Object {
          "flex": 1,
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          },
          false,
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#000",
            "height": 0,
            "width": "100%",
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "paddingHorizontal": 16,
              "width": "100%",
            },
            Object {
              "backgroundColor": "#000",
              "marginBottom": -0,
              "paddingBottom": 0,
            },
            Object {},
            Object {},
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "flex-end",
              "opacity": 1,
            }
          }
          testID="dismissable-background"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 3,
                "justifyContent": "center",
                "marginTop": 64,
              }
            }
          >
            <View
              style={
                Object {
                  "justifyContent": "flex-end",
                  "marginBottom": 72,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "paddingTop": 5,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "rgba(255,255,255,0.85)",
                      "fontFamily": "TTCommons-Medium",
                      "fontSize": 28,
                      "letterSpacing": 0,
                      "lineHeight": 32,
                    },
                    undefined,
                  ]
                }
              >
                AusweisCompatibilityStatus.header
              </Text>
            </View>
            <View
              style={
                Object {
                  "justifyContent": "flex-start",
                  "paddingHorizontal": 52,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "paddingTop": 5,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#ffcc01",
                      "fontFamily": "TTCommons-Regular",
                      "fontSize": 22,
                      "letterSpacing": 0.15,
                      "lineHeight": 26,
                    },
                    undefined,
                  ]
                }
              >
                AusweisCompatibilityStatus.error
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "rgba(255,255,255,0.9)",
                  "borderRadius": 34,
                  "borderWidth": 3,
                  "height": 67,
                  "justifyContent": "center",
                  "marginBottom": 20,
                  "padding": 17,
                  "width": 67,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="100%"
                bbWidth="100%"
                focusable={false}
                height="100%"
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "flex": 0,
                      "height": "100%",
                      "width": "100%",
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
                width="100%"
              >
                <RNSVGGroup>
                  <RNSVGPath
                    d="M10.852 12.004L.23 22.7a.764.764 0 000 1.076.75.75 0 001.068 0L12 13l10.703 10.778a.75.75 0 001.069 0 .764.764 0 000-1.076L13.148 12.004l10.63-10.706a.764.764 0 000-1.075.752.752 0 00-1.067 0L12 11.008 1.289.223a.752.752 0 00-1.068 0 .764.764 0 000 1.076l10.631 10.705z"
                    fill={3875536895}
                    fillRule={1}
                    propList={
                      Array [
                        "fill",
                        "fillRule",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "paddingTop": 5,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "rgba(255, 255, 255, 0.8)",
                    "fontFamily": "TTCommons-Regular",
                    "fontSize": 22,
                    "letterSpacing": 0.15,
                    "lineHeight": 26,
                  },
                  undefined,
                ]
              }
            >
              AusweisCompatibilityStatus.error
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`Ausweis compatibility result screen is according to the designs when: scanned card is fully supported 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
        Object {
          "flex": 1,
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          },
          false,
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#000",
            "height": 0,
            "width": "100%",
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "paddingHorizontal": 16,
              "width": "100%",
            },
            Object {
              "backgroundColor": "#000",
              "marginBottom": -0,
              "paddingBottom": 0,
            },
            Object {},
            Object {},
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "flex-end",
              "opacity": 1,
            }
          }
          testID="dismissable-background"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 3,
                "justifyContent": "center",
                "marginTop": 64,
              }
            }
          >
            <View
              style={
                Object {
                  "justifyContent": "flex-end",
                  "marginBottom": 72,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "paddingTop": 5,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "rgba(255,255,255,0.85)",
                      "fontFamily": "TTCommons-Medium",
                      "fontSize": 28,
                      "letterSpacing": 0,
                      "lineHeight": 32,
                    },
                    undefined,
                  ]
                }
              >
                AusweisCompatibilityStatus.header
              </Text>
            </View>
            <View
              style={
                Object {
                  "justifyContent": "flex-start",
                  "paddingHorizontal": 52,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "paddingVertical": 16,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "paddingTop": 5,
                        "textAlign": "center",
                      },
                      Object {
                        "color": "rgba(255, 255, 255, 0.8)",
                        "fontFamily": "TTCommons-Regular",
                        "fontSize": 22,
                        "letterSpacing": 0.15,
                        "lineHeight": 26,
                      },
                      undefined,
                    ]
                  }
                >
                  AusweisCompatibilityStatus.status1
                </Text>
                <View
                  style={
                    Object {
                      "height": 20,
                      "marginTop": 16,
                      "width": 20,
                    }
                  }
                >
                  <RNSVGSvgView
                    align="xMidYMid"
                    bbHeight="100%"
                    bbWidth="100%"
                    focusable={false}
                    height="100%"
                    meetOrSlice={0}
                    minX={0}
                    minY={0}
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "borderWidth": 0,
                        },
                        Object {
                          "flex": 0,
                          "height": "100%",
                          "width": "100%",
                        },
                      ]
                    }
                    vbHeight={20}
                    vbWidth={20}
                    width="100%"
                  >
                    <RNSVGGroup>
                      <RNSVGGroup
                        fill={null}
                        fillRule={1}
                        propList={
                          Array [
                            "fill",
                            "fillRule",
                          ]
                        }
                      >
                        <RNSVGPath
                          d="M17.071 2.929A9.935 9.935 0 0010 0a9.935 9.935 0 00-7.071 2.929A9.935 9.935 0 000 10a9.934 9.934 0 002.929 7.071A9.935 9.935 0 0010 20a9.935 9.935 0 007.071-2.929A9.935 9.935 0 0020 10a9.935 9.935 0 00-2.929-7.071z"
                          fill={4283584961}
                          propList={
                            Array [
                              "fill",
                            ]
                          }
                        />
                        <RNSVGPath
                          d="M14.727 7.172a.586.586 0 00-.829 0l-5.174 5.174L6 9.622a.586.586 0 00-.828.829l3.138 3.138a.584.584 0 00.828 0L14.727 8a.586.586 0 000-.828z"
                          fill={4294967295}
                          propList={
                            Array [
                              "fill",
                            ]
                          }
                        />
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGSvgView>
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "paddingVertical": 16,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "paddingTop": 5,
                        "textAlign": "center",
                      },
                      Object {
                        "color": "rgba(255, 255, 255, 0.8)",
                        "fontFamily": "TTCommons-Regular",
                        "fontSize": 22,
                        "letterSpacing": 0.15,
                        "lineHeight": 26,
                      },
                      undefined,
                    ]
                  }
                >
                  AusweisCompatibilityStatus.status2
                </Text>
                <View
                  style={
                    Object {
                      "height": 20,
                      "marginTop": 16,
                      "width": 20,
                    }
                  }
                >
                  <RNSVGSvgView
                    align="xMidYMid"
                    bbHeight="100%"
                    bbWidth="100%"
                    focusable={false}
                    height="100%"
                    meetOrSlice={0}
                    minX={0}
                    minY={0}
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "borderWidth": 0,
                        },
                        Object {
                          "flex": 0,
                          "height": "100%",
                          "width": "100%",
                        },
                      ]
                    }
                    vbHeight={20}
                    vbWidth={20}
                    width="100%"
                  >
                    <RNSVGGroup>
                      <RNSVGGroup
                        fill={null}
                        fillRule={1}
                        propList={
                          Array [
                            "fill",
                            "fillRule",
                          ]
                        }
                      >
                        <RNSVGPath
                          d="M17.071 2.929A9.935 9.935 0 0010 0a9.935 9.935 0 00-7.071 2.929A9.935 9.935 0 000 10a9.934 9.934 0 002.929 7.071A9.935 9.935 0 0010 20a9.935 9.935 0 007.071-2.929A9.935 9.935 0 0020 10a9.935 9.935 0 00-2.929-7.071z"
                          fill={4283584961}
                          propList={
                            Array [
                              "fill",
                            ]
                          }
                        />
                        <RNSVGPath
                          d="M14.727 7.172a.586.586 0 00-.829 0l-5.174 5.174L6 9.622a.586.586 0 00-.828.829l3.138 3.138a.584.584 0 00.828 0L14.727 8a.586.586 0 000-.828z"
                          fill={4294967295}
                          propList={
                            Array [
                              "fill",
                            ]
                          }
                        />
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGSvgView>
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "paddingVertical": 16,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "paddingTop": 5,
                        "textAlign": "center",
                      },
                      Object {
                        "color": "rgba(255, 255, 255, 0.8)",
                        "fontFamily": "TTCommons-Regular",
                        "fontSize": 22,
                        "letterSpacing": 0.15,
                        "lineHeight": 26,
                      },
                      undefined,
                    ]
                  }
                >
                  AusweisCompatibilityStatus.status3
                </Text>
                <View
                  style={
                    Object {
                      "height": 20,
                      "marginTop": 16,
                      "width": 20,
                    }
                  }
                >
                  <RNSVGSvgView
                    align="xMidYMid"
                    bbHeight="100%"
                    bbWidth="100%"
                    focusable={false}
                    height="100%"
                    meetOrSlice={0}
                    minX={0}
                    minY={0}
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "borderWidth": 0,
                        },
                        Object {
                          "flex": 0,
                          "height": "100%",
                          "width": "100%",
                        },
                      ]
                    }
                    vbHeight={20}
                    vbWidth={20}
                    width="100%"
                  >
                    <RNSVGGroup>
                      <RNSVGGroup
                        fill={null}
                        fillRule={1}
                        propList={
                          Array [
                            "fill",
                            "fillRule",
                          ]
                        }
                      >
                        <RNSVGPath
                          d="M17.071 2.929A9.935 9.935 0 0010 0a9.935 9.935 0 00-7.071 2.929A9.935 9.935 0 000 10a9.934 9.934 0 002.929 7.071A9.935 9.935 0 0010 20a9.935 9.935 0 007.071-2.929A9.935 9.935 0 0020 10a9.935 9.935 0 00-2.929-7.071z"
                          fill={4283584961}
                          propList={
                            Array [
                              "fill",
                            ]
                          }
                        />
                        <RNSVGPath
                          d="M14.727 7.172a.586.586 0 00-.829 0l-5.174 5.174L6 9.622a.586.586 0 00-.828.829l3.138 3.138a.584.584 0 00.828 0L14.727 8a.586.586 0 000-.828z"
                          fill={4294967295}
                          propList={
                            Array [
                              "fill",
                            ]
                          }
                        />
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGSvgView>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "rgba(255,255,255,0.9)",
                  "borderRadius": 34,
                  "borderWidth": 3,
                  "height": 67,
                  "justifyContent": "center",
                  "marginBottom": 20,
                  "padding": 17,
                  "width": 67,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="100%"
                bbWidth="100%"
                focusable={false}
                height="100%"
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "flex": 0,
                      "height": "100%",
                      "width": "100%",
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={32}
                width="100%"
              >
                <RNSVGGroup>
                  <RNSVGPath
                    d="M31.333.357a1.219 1.219 0 00-1.724 0L9.852 20.114 2.08 12.343a1.219 1.219 0 10-1.724 1.724L8.99 22.7a1.22 1.22 0 001.724 0L31.333 2.08a1.219 1.219 0 000-1.723z"
                    fill={3875536895}
                    fillRule={1}
                    propList={
                      Array [
                        "fill",
                        "fillRule",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "paddingTop": 5,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "rgba(255, 255, 255, 0.8)",
                    "fontFamily": "TTCommons-Regular",
                    "fontSize": 22,
                    "letterSpacing": 0.15,
                    "lineHeight": 26,
                  },
                  undefined,
                ]
              }
            >
              AusweisCompatibilityStatus.success
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`Ausweis compatibility result screen is according to the designs when: scanned card is inopeartive 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
        Object {
          "flex": 1,
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          },
          false,
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#000",
            "height": 0,
            "width": "100%",
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "paddingHorizontal": 16,
              "width": "100%",
            },
            Object {
              "backgroundColor": "#000",
              "marginBottom": -0,
              "paddingBottom": 0,
            },
            Object {},
            Object {},
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "flex-end",
              "opacity": 1,
            }
          }
          testID="dismissable-background"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 3,
                "justifyContent": "center",
                "marginTop": 64,
              }
            }
          >
            <View
              style={
                Object {
                  "justifyContent": "flex-end",
                  "marginBottom": 72,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "paddingTop": 5,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "rgba(255,255,255,0.85)",
                      "fontFamily": "TTCommons-Medium",
                      "fontSize": 28,
                      "letterSpacing": 0,
                      "lineHeight": 32,
                    },
                    undefined,
                  ]
                }
              >
                AusweisCompatibilityStatus.header
              </Text>
            </View>
            <View
              style={
                Object {
                  "justifyContent": "flex-start",
                  "paddingHorizontal": 52,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "paddingTop": 5,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#ffcc01",
                      "fontFamily": "TTCommons-Regular",
                      "fontSize": 22,
                      "letterSpacing": 0.15,
                      "lineHeight": 26,
                    },
                    undefined,
                  ]
                }
              >
                AusweisCompatibilityStatus.error
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "rgba(255,255,255,0.9)",
                  "borderRadius": 34,
                  "borderWidth": 3,
                  "height": 67,
                  "justifyContent": "center",
                  "marginBottom": 20,
                  "padding": 17,
                  "width": 67,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="100%"
                bbWidth="100%"
                focusable={false}
                height="100%"
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "flex": 0,
                      "height": "100%",
                      "width": "100%",
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
                width="100%"
              >
                <RNSVGGroup>
                  <RNSVGPath
                    d="M10.852 12.004L.23 22.7a.764.764 0 000 1.076.75.75 0 001.068 0L12 13l10.703 10.778a.75.75 0 001.069 0 .764.764 0 000-1.076L13.148 12.004l10.63-10.706a.764.764 0 000-1.075.752.752 0 00-1.067 0L12 11.008 1.289.223a.752.752 0 00-1.068 0 .764.764 0 000 1.076l10.631 10.705z"
                    fill={3875536895}
                    fillRule={1}
                    propList={
                      Array [
                        "fill",
                        "fillRule",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "paddingTop": 5,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "rgba(255, 255, 255, 0.8)",
                    "fontFamily": "TTCommons-Regular",
                    "fontSize": 22,
                    "letterSpacing": 0.15,
                    "lineHeight": 26,
                  },
                  undefined,
                ]
              }
            >
              AusweisCompatibilityStatus.error
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
